sci_strategic_plugin/src/sci_strategic_plugin.cpp

```diff
@@ -86,63 +86,90 @@ void SCIStrategicPlugin::mobilityOperationCb(const cav_msgs::MobilityOperationCo
+        parseStrategyParams(msg->strategy_params);
+      }
+       
     }
     previous_strategy_params_ = msg->strategy_params;
-
-    generateMobilityOperation();
   }
   
 }
 
 void SCIStrategicPlugin::BSMCb(const cav_msgs::BSMConstPtr& msg)
 {
-  bsm_id = msg->core_data.id;
+  std::vector<uint8_t> bsm_id_vec = msg->core_data.id;
+
+  std::string id = "";
+  for (size_t i=0; i< bsm_id_vec.size(); i++)
+  {
+    id += std::to_string(bsm_id_vec[i]);
+  }
+
+  bsm_id_ = id;
+  bsm_msg_count_ = msg->core_data.msg_count;
+  bsm_sec_mark_ = msg->core_data.sec_mark;
 }
 
 void SCIStrategicPlugin::currentPoseCb(const geometry_msgs::PoseStampedConstPtr& msg)
 {
   geometry_msgs::PoseStamped pose_msg = geometry_msgs::PoseStamped(*msg.get());
-  lanelet::BasicPoint2d current_loc(pose_msg.pose.position.x, pose_msg.pose.position.y);
-  current_downtrack_ = wm_->routeTrackPos(current_loc).downtrack;
-  ROS_DEBUG_STREAM("Downtrack from current pose: " << current_downtrack_);
+  if (vehicle_engaged_)
+  {
+    lanelet::BasicPoint2d current_loc(pose_msg.pose.position.x, pose_msg.pose.position.y);
+    current_downtrack_ = wm_->routeTrackPos(current_loc).downtrack;
+    ROS_DEBUG_STREAM("Downtrack from current pose: " << current_downtrack_);
+  }
+  
+  
 }
 
 void SCIStrategicPlugin::parseStrategyParams(const std::string& strategy_params)
 {
-  std::istringstream strategy_params_ss(strategy_params);
-  boost::property_tree::ptree parser;
-  boost::property_tree::ptree child;
-  boost::property_tree::json_parser::read_json(strategy_params_ss, parser);
-  child = parser.get_child("schedule_plan");
-  for(const auto& p : child)
-  {
-    if (p.first == "metadata")
-    {
-      street_msg_timestamp_ = p.second.get<uint32_t>("timestamp");
-    }
-    if (p.first == "payload" && p.second.get<std::string>("veh_id") == config_.vehicle_id)
-    {        
-      // parse stop time in ms
-      scheduled_stop_time_ = p.second.get<uint32_t>("est_stop_t");
-      ROS_DEBUG_STREAM("scheduled_stop_time_: " << scheduled_stop_time_);
-
-      scheduled_enter_time_ = p.second.get<uint32_t>("est_enter_t");
-      ROS_DEBUG_STREAM("scheduled_enter_time_: " << scheduled_enter_time_);
-      
-      scheduled_depart_time_ = p.second.get<uint32_t>("est_depart_t");
-      ROS_DEBUG_STREAM("scheduled_depart_time_: " << scheduled_depart_time_);
-
-      scheduled_latest_depart_time_ = p.second.get<uint32_t>("latest_depart_p");
-      ROS_DEBUG_STREAM("scheduled_latest_depart_time_: " << scheduled_latest_depart_time_);
+  // sample strategy_params: "st:1634067044,et:1634067059, dt:1634067062.3256602,dp:2,,access: 0"
+  std::string params = strategy_params;
+  std::vector<std::string> inputsParams;
+  boost::algorithm::split(inputsParams, params, boost::is_any_of(","));
+
+  std::vector<std::string> st_parsed;
+  boost::algorithm::split(st_parsed, inputsParams[0], boost::is_any_of(":"));
+  uint32_t st = std::stoi(st_parsed[1]);
```

Comment:
Is there a reason for this intermediate variable in all of these parsings, instead of just:
`scheduled_stop_time_ = std::stoi(st_parsed[1])`?